### PR TITLE
Add stress tool cpu usage to output

### PR DESCRIPTION
**Pull request by arun-koshy**
```
arunkoshy@MystenLoshysMBP sui-benchmark % cargo run --package sui-benchmark --bin stress -- --log-path /tmp/stress.log --num-client-threads 10 --num-server-threads 24 --num-transfer-accounts 2 bench --target-qps 100 --num-workers 10  --shared-counter 100 --run-duration 30s --stress-stat-collection true

Benchmark Report:
+-------------+-----+--------+---------------+---------------+---------------+
| duration(s) | tps | error% | latency (min) | latency (p50) | latency (p99) |
+============================================================================+
| 30          | 98  | 0      | 91            | 456           | 778           |
+-------------+-----+--------+---------------+---------------+---------------+
Stress Performance Report:
+-----------+-----+-----+
| metric    | p50 | p99 |
+=======================+
| cpu usage | 39  | 64  |
+-----------+-----+-----+


arunkoshy@MystenLoshysMBP sui-benchmark % cargo run --package sui-benchmark --bin stress -- --log-path /tmp/stress.log --num-client-threads 10 --num-server-threads 12 --num-transfer-accounts 2 bench --target-qps 200 --num-workers 10  --shared-counter 200 --run-duration 30s --stress-stat-collection true
Benchmark Report:
+-------------+-----+--------+---------------+---------------+---------------+
| duration(s) | tps | error% | latency (min) | latency (p50) | latency (p99) |
+============================================================================+
| 30          | 196 | 0      | 196           | 519           | 870           |
+-------------+-----+--------+---------------+---------------+---------------+
Stress Performance Report:
+-----------+-----+-----+
| metric    | p50 | p99 |
+=======================+
| cpu usage | 77  | 97  |
+-----------+-----+-----+
```

### CODE REVIEW

File: crates/sui-benchmark/src/drivers/mod.rs [hunk 1]
```diff
@@ -119,6 +131,38 @@ impl BenchmarkStats {
         table.add_row(row);
         table
     }
+
+    pub fn to_cpu_table(&self) -> Table {
+        let mut table = Table::new();
+        table
```
Comment (mwtian) [inline crates/sui-benchmark/src/drivers/mod.rs:137]: nit: I think we don't need this many quantiles for CPU or latency. For CPU just average would be good enough. For latency, min, p50 and p99 should be good enough.

Per-core CPU info may be a bit too much as well. Is it easy to calculate avg CPU across all cores over the period, and move CPU and latency to a single table?

**Review by mwtian — COMMENTED** [review]
Great, looking forward to this!

File: crates/sui-benchmark/src/drivers/bench_driver.rs [hunk 2]
```diff
@@ -532,7 +589,52 @@ impl Driver<BenchmarkStats> for BenchDriver {
...
         };
-        let benchmark_stat = stat_task.await.unwrap();
-        Ok(benchmark_stat)
+        let benchmark_stat = benchmark_stat_task.await.unwrap();
+        let stress_stat = stress_stat_task.await.unwrap();
+        Ok((benchmark_stat, stress_stat))
     }
 }
+
+fn stress_stats_collector(
```
Comment (mwtian) [inline crates/sui-benchmark/src/drivers/bench_driver.rs:598]: Maybe have a high level description of how this works, e.g. collection interval, when will this end, etc

**Review by mwtian — APPROVED** [review]

File: crates/sui-benchmark/src/options.rs [hunk 3]
```diff
@@ -99,6 +99,9 @@ pub struct Opts {
     // Stat collection interval seconds
     #[clap(long, default_value = "10", global = true)]
     pub stat_collection_interval: u64,
+    // Enable stress stat collection
+    #[clap(long, parse(try_from_str), default_value = "false", global = true)]
+    pub stress_stat_collection: bool,
```
Comment (mwtian) [inline crates/sui-benchmark/src/options.rs:104]: What is the downside of always enabling this?

Comment (arun-koshy) [reply crates/sui-benchmark/src/options.rs:104]: It [breaks](https://github.com/MystenLabs/sui/actions/runs/3942663537/jobs/6746580409) the sim test, as per marks suggestion it was best to disable it in the tests.

**Pull request merged**